base_local_planner/src/critics/goal_heading_cost_function.cpp

```diff
@@ -0,0 +1,79 @@
+/*********************************************************************
+ *
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2016, 6 River Systems
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage, Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Daniel Grieneisen
+ *********************************************************************/
+
+#include <base_local_planner/critics/goal_heading_cost_function.h>
+#include <tf/transform_datatypes.h>
+
+namespace base_local_planner {
+
+GoalHeadingCostFunction::GoalHeadingCostFunction()  {
+    // nothing to set yet
+}
+
+bool GoalHeadingCostFunction::prepare() {
+
+  // pre calculate goal heading
+ goal_heading_ = tf::getYaw(goal_pose_.pose.orientation);
+
+  return true;
+}
+
+double GoalHeadingCostFunction::scoreTrajectory(Trajectory &traj) {
+  double path_end_x, path_end_y, path_end_th;
+  traj.getEndpoint(path_end_x, path_end_y, path_end_th); // what units is th??
+
+  double heading_diff = path_end_th - goal_heading_;
+  // Normalize the heading difference
+  while (heading_diff > M_PI)
```

Comment:
We probably have a helper function for this somewhere, but no reason not to implement it here.